flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -304,6 +322,11 @@ public void testThrottling() throws Exception {
 
     try (CloseableIterator<RowData> resultIterator = stream.collectAsync()) {
       JobClient jobClient = env.executeAsync("Iceberg Source Throttling Test");
+      CommonTestUtils.waitForAllTaskRunning(
```

Comment:
is this the main fix to make sure all tasks running first?

other code refactoring/reordering is just cosmetic/style thing?

Reply:
I needed to reorder the Iceberg commit after starting the job. So the first records are moved. To make the test more consistent, moved the record generation to the beginning of the test